drivers/exec/driver.go

```diff
@@ -177,20 +178,70 @@ func (d *ExecDriver) handleFingerprint(ctx context.Context, ch chan *drivers.Fin
 	}
 }
 
-func (d *ExecDriver) RecoverTask(handle *drivers.TaskHandle) error {
-	var taskState TaskState
-	logger := d.logger.With("task_id", handle.Config.ID)
+func (d *Driver) buildFingerprint() *drivers.Fingerprint {
+	if runtime.GOOS != "linux" {
+		return &drivers.Fingerprint{
+			Health:            drivers.HealthStateUndetected,
+			HealthDescription: "exec driver unsupported on client OS",
+		}
+	}
+
+	fp := &drivers.Fingerprint{
+		Attributes:        map[string]string{},
+		Health:            drivers.HealthStateHealthy,
+		HealthDescription: "healthy",
+	}
 
-	err := handle.GetDriverState(&taskState)
+	mount, err := fingerprint.FindCgroupMountpointDir()
 	if err != nil {
-		logger.Error("failed to decode driver state during task recovery", "error", err)
-		return fmt.Errorf("failed to decode state: %v", err)
+		fp.Health = drivers.HealthStateUnhealthy
+		fp.HealthDescription = "failed to discover cgroup mount point"
+		d.logger.Warn(fp.HealthDescription, "error", err)
+		return fp
+	}
+
+	if mount == "" {
+		fp.Health = drivers.HealthStateUnhealthy
+		fp.HealthDescription = "cgroups are unavailable"
+		return fp
+	}
+
+	if unix.Geteuid() != 0 {
```

Comment:
helper seems reasonable, but also I'd like to avoid returning an `error` and I cannot think of a sensible value for Windows - let me chew on it and follow up.  Maybe introduce a `IsUnixRoot()` or `IsRoot()`?